x-pack/plugins/actions/server/usage/connector_usage_reporting_task.ts

```diff
@@ -0,0 +1,309 @@
+    this.logger.debug(`Record: ${JSON.stringify(record)}`);
+
+    try {
+      attempts = attempts + 1;
+      await this.pushUsageRecord(record);
+      this.logger.info(
+        `Connector usage record has been successfully reported, ${record.creation_timestamp}, usage: ${record.usage.quantity}, period:${record.usage.period_seconds}`
+      );
+    } catch (e) {
+      if (attempts < MAX_PUSH_ATTEMPTS) {
+        this.logger.error(
+          `Usage data could not be pushed to usage-api. It will be retried (${attempts}). Error:${e.message}`
+        );
+
+        return {
+          state: {
+            lastReportedUsageDate,
+            attempts,
+          },
+          runAt: this.getDelayedRetryDate({ attempts, now }),
+        };
+      }
+      this.logger.error(
+        `Usage data could not be pushed to usage-api. Stopped retrying after ${attempts} attempts. Error:${e.message}`
+      );
+      return {
+        state: {
+          lastReportedUsageDate,
+          attempts: 0,
+        },
+      };
+    }
+
+    return {
+      state: { lastReportedUsageDate: toDate, attempts: 0 },
+    };
+  };
+
+  private getTotalUsage = async ({
+    esClient,
+    fromDate,
+    toDate,
+  }: {
+    esClient: ElasticsearchClient;
+    fromDate: Date;
+    toDate: Date;
+  }): Promise<number> => {
+    const usageResult = await esClient.search({
+      index: this.eventLogIndex,
+      sort: '@timestamp',
+      size: 0,
+      query: {
+        bool: {
+          filter: {
+            bool: {
+              must: [
+                {
+                  term: { 'event.action': 'execute' },
+                },
+                {
+                  term: { 'event.provider': 'actions' },
+                },
+                {
+                  exists: {
+                    field: 'kibana.action.execution.usage.request_body_bytes',
+                  },
+                },
+                {
+                  range: {
+                    '@timestamp': {
+                      gt: fromDate,
+                      lte: toDate,
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        },
+      },
+      aggs: {
+        total_usage: { sum: { field: 'kibana.action.execution.usage.request_body_bytes' } },
+      },
+    });
+
+    return (usageResult.aggregations?.total_usage as AggregationsSumAggregate)?.value ?? 0;
+  };
+
+  private createUsageRecord = ({
+    totalUsage,
+    fromDate,
+    toDate,
+    projectId,
+  }: {
+    totalUsage: number;
+    fromDate: Date;
+    toDate: Date;
+    projectId: string;
+  }): ConnectorUsageReport => {
+    const period = Math.round((toDate.getTime() - fromDate.getTime()) / 1000);
+    const toStr = toDate.toISOString();
+    const timestamp = new Date(toStr);
+    timestamp.setMinutes(0);
+    timestamp.setSeconds(0);
+    timestamp.setMilliseconds(0);
+
+    return {
+      id: `connector-request-body-bytes-${timestamp.toISOString()}`,
```

Comment:
You might want to consider making this more unique. `id` needs to be _globally_ unique, so adding in `project_id` or something similar could help ensure that